### ISSUE TITLE
Executables not being built for CEGUI sample code

### ISSUE THREAD

**Issue by Bluemoon1972** [cegui/cegui#1301]
I used CMake and VS2022 to build the latest version of CEGUI (and its dependencies). Everything seemed to be in order, and I checked to make sure that CEGUI_SAMPLE_ENABLED and the individual project variables were all enabled so that CMake would generate buildable samples for VS2022.

I can open and build the samples solution in VS 2022 without any problems, but I can't run any of the projects in Debug mode. Instead it complains that the relevant .dll is not a valid Win32 application. 
I also tried to run the Release executable, but the Release directories don't have any executables in them.

So clearly the executable files for Debug and Release haven't been created. I went through the entire process again, but this time building a Win32 configuration for VS2017, instead of x64 for VS2022. Same problem - no executables being built for sample projects for either Debug or Release.

I have looked at the CEGUI online documentation and various Youtube videos to find out how to run the samples, but have found nothing beyond a couple of passing references which suggest that they are ready to run once CEGUI is built.

**Comment by Ident** [comment]
Did you placee the dependencies into the dependencies folder after building them?

**Comment by Bluemoon1972** [comment]
CEGUI-0.8.7 is the version I am working with.

I moved the dependencies folder created by the dependencies build into the main directory for CEGUI. Presumably the CEGUI build would have simply failed without that being present.

**Comment by Ident** [comment]
Yes it would have. Unfortunately i dont have a set up v0-8 version on my PC right now to test it.

> Instead it complains that the relevant .dll is not a valid Win32 application.
What is the error exactly?

**Comment by Bluemoon1972** [comment]
![image_2023-11-10_150448948](https://github.com/cegui/cegui/assets/150435300/0aa0196a-f9b6-413d-b335-e0f5b2858322)

**Comment by Ident** [comment]
> If there is a more widely used or easier to check version, I can certainly give that a try instead
I would use default branch. It has a sample browser instead of executables for each sample so it behaves a bit differently.

> What is the sample browser?
It might not yet exist in 0.8.7. I created it like 10 years ago lol but 0.8.X is quite old and kept stable.

**Comment by Bluemoon1972** [comment]
> I would use default branch. It has a sample browser instead of executables for each sample so it behaves a bit differently.

Okay, thank you. Perhaps I will try that version.

> Regarding your image: It looks like it builds them as dlls so you cant start them directly.

Right, I suppose it makes sense that the default CMake output would set Visual Studio to produce dlls for CEGUI. So I guess I need to change my configuration in Visual Studio. Presumably that is why my Release build generates nothing at all. I shall try to get it outputting and executable.

**Comment by Bluemoon1972** [comment]
I tried switching the output from .dll to .exe and now I am getting fatal linking errors. It looks like it is quite difficult to change the output type once a Visual Studio project has been created, so I think I will be better off trying the default version of CEGUI, since that is newer.

Thank you for me pointing in a more productive direction.

**Comment by Ident** [comment]
> Right, I suppose it makes sense that the default CMake output would set Visual Studio to produce CEGUI dlls. So I guess I need to change my configuration in Visual Studio. Presumably that is why my Release build generates nothing at all. I shall try to get it outputting an executable.

hm actually i d blame CEGUI setup in this case. CMAKE should set the project files up in a way you get an exe or dll depending on whether it should be an exe or dll in the end. We might have screwed this up for v0-8 at some point :/ i dont know
